lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimConfigurationProvider.java

```diff
@@ -32,10 +38,23 @@
 public class AzureBatchEvaluatorShimConfigurationProvider {
 
   private final RemoteManager remoteManager;
+  private final TcpPortProvider portProvider;
+  private final LocalAddressProvider localAddressProvider;
+  private final Set<String> tcpPortSet;
 
   @Inject
-  AzureBatchEvaluatorShimConfigurationProvider(final RemoteManager remoteManager) {
+  AzureBatchEvaluatorShimConfigurationProvider(
+      @Parameter(TcpPortSet.class) final Set<Integer> tcpPortSet,
+      final RemoteManager remoteManager,
+      final LocalAddressProvider localAddressProvider,
+      final TcpPortProvider portProvider) {
     this.remoteManager = remoteManager;
+    this.portProvider = portProvider;
+    this.localAddressProvider = localAddressProvider;
```

Comment:
We have a portProvider and the tcpPortSet; shouldn't there only be one or the other?

Reply:
We pass on the portSet to the shim but also the portprovider implementation since that allows the shim to operate with whatever provider the driver got. This is needed so that this same code can work for both container-based (the set is used along with the Set TcpPortProvider implementation) as well as non-container based workloads (the default range provider is used).

---
In reply to: [201792204](https://github.com/sharathmalladi/reef/pull/5#discussion_r201792204) [](ancestors = 201792204)

Reply:
Why would this be an if/else scenario since both the Container and Non-Container implementations set the Port list using the same parameter?

---
In reply to: [201805509](https://github.com/sharathmalladi/reef/pull/5#discussion_r201805509) [](ancestors = 201805509,201792204)

Reply:
Resolved by setting shim configuration separately instead of borrowing from driver configuration.

---
In reply to: [201805509](https://github.com/sharathmalladi/reef/pull/5#discussion_r201805509) [](ancestors = 201805509,201792204)